ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -99,24 +124,20 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDef := &ecs.ContainerDefinition{
 			Name: aws.String(name),
 		}
-		if err := convertToContainerDef(context, serviceConfig, volumes, containerDef); err != nil {
-			return nil, err
+
+		// Check if essential is specified for the container
+		ecsContainerDef := &ContainerDef{}
+		if cd, ok := ecsContainerDefs[name]; ok {
+			ecsContainerDef = &cd
 		}
-		containerDefinitions = append(containerDefinitions, containerDef)
-	}
 
-	ecsParams, err := readECSParams(ecsParamsFileName)
-	if err != nil {
-		return nil, err
-	}
+		allEssential := areContainersAllEssential(ecsContainerDefs)
```

Comment:
So in the old CLI, we didn't have essential, and so field was auto-set to true by control plane. 

As I understand your code, if the person doesn't specify essential for any containers then all are still set to true.

BUT, what if I set it to be false for only one container? Ie, what if I have 10 service definitions in my docker-compose file, but I have an ECS Params file like this:
```
services:
     mysql:
          essential false
```

And my docker compose file is like this:
```
version: "2"
services:
  web:
    build: web
    command: python app.py
    ports:
     - "5000:5000"
    volumes:
     - ./web:/code
    links:
     - redis
    environment:
  redis:
    image: redis
  # agent section
  mysql:
    image: mysql
    links:
     - web
```

If I understand your code correctly, this would end up setting all of the containers to have essential be false. This is potentially fine- I don't think its a bug. But in a way you have changed the default for essential from True to False- I think we should talk about that in person.

Reply:
That's a great point -- I hadn't considered that edge case. It's true that this would set all containers to be false, and you would see an error like this:

```
~/sandbox/test_fields$ ecs-cli compose up --cluster test-ecs-fields
WARN[0000] Skipping unsupported YAML option...           option name=networks
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=mysql
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=wordpress
ERRO[0000] Error registering task definition             error="ClientException: Task definition doesn't have any essential container.\n\tstatus code: 400, request id: cc373549-b46b-11e7-8f7f-e3294f673e50" family="ecscompose-test_fields"
ERRO[0000] Create task definition failed                 error="ClientException: Task definition doesn't have any essential container.\n\tstatus code: 400, request id: cc373549-b46b-11e7-8f7f-e3294f673e50"
FATA[0000] ClientException: Task definition doesn't have any essential container.
	status code: 400, request id: cc373549-b46b-11e7-8f7f-e3294f673e50
```

So you're right in that it's not a bug per se, but it's not a great user experience and is a good argument for making the `areContainersAllEssential` function less trivial =P